### PR TITLE
Update dependencies (29-03-21)

### PR DESCRIPTION
**Pull request by veado**
Some "highlights":

- [x] Latest `typescript@4.2.x`
- [x] Add config files for latest `husky`
- [x] Update config for `eslint` 
- [x] Fix `fp-ts` dependencies in `yarn.lock`
- [x] Remove `bitcore-* deps`

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -467,6 +478,11 @@
   resolved "https://registry.yarnpkg.com/@babel/helper-validator-option/-/helper-validator-option-7.12.1.tgz#175567380c3e77d60ff98a54bb015fe78f2178d9"
   integrity sha512-YpJabsXlJVWP0USHjnC/AQDTLlZERbON577YUVO/wLpqyj6HAtVYnWaQaN0iUN+1/tWn3c+uKKXjRut5115Y2A==
 
+"@babel/helper-validator-option@^7.12.17":
+  version "7.12.17"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:482]: seems this is a duplicated version

File: yarn.lock [hunk 2]
```diff
@@ -723,6 +739,13 @@
   dependencies:
     "@babel/helper-plugin-utils" "^7.10.4"
 
+"@babel/plugin-syntax-jsx@^7.12.13":
+  version "7.12.13"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:743]: this looks duplicated too

File: yarn.lock [hunk 3]
```diff
@@ -1017,6 +1047,13 @@
...
 
+"@babel/plugin-transform-react-jsx-development@^7.12.17":
+  version "7.12.17"
+  resolved "https://registry.yarnpkg.com/@babel/plugin-transform-react-jsx-development/-/plugin-transform-react-jsx-development-7.12.17.tgz#f510c0fa7cd7234153539f9a362ced41a5ca1447"
+  integrity sha512-BPjYV86SVuOaudFhsJR1zjgxxOhJDt6JHNoD48DxWEIxUCAMjV1ys6DYw4SDYZh0b1QsS2vfIA9t/ZsQGsDOUQ==
+  dependencies:
+    "@babel/plugin-transform-react-jsx" "^7.12.17"
+
 "@babel/plugin-transform-react-jsx-development@^7.12.7":
   version "7.12.11"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:1058]: this looks duplicated too

File: yarn.lock [hunk 4]
```diff
@@ -1060,6 +1097,17 @@
     "@babel/helper-plugin-utils" "^7.10.4"
     "@babel/plugin-syntax-jsx" "^7.12.1"
 
+"@babel/plugin-transform-react-jsx@^7.12.17", "@babel/plugin-transform-react-jsx@^7.13.12":
+  version "7.13.12"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:1101]: this looks duplicated too

File: yarn.lock [hunk 5]
```diff
@@ -1415,6 +1462,15 @@
     lodash "^4.17.19"
     to-fast-properties "^2.0.0"
 
+"@babel/types@^7.12.13", "@babel/types@^7.13.12":
+  version "7.13.14"
+  resolved "https://registry.yarnpkg.com/@babel/types/-/types-7.13.14.tgz#c35a4abb15c7cd45a2746d78ab328e362cbace0d"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:1467]: this looks duplicated too

File: yarn.lock [hunk 6]
```diff
@@ -1579,10 +1635,10 @@
     "@emotion/sheet" "0.9.4"
     "@emotion/utils" "0.11.3"
 
-"@emotion/core@^10.0.9":
-  version "10.0.28"
-  resolved "https://registry.yarnpkg.com/@emotion/core/-/core-10.0.28.tgz#bb65af7262a234593a9e952c041d0f1c9b9bef3d"
-  integrity sha512-pH8UueKYO5jgg0Iq+AmCLxBsvuGtvlmiDCOuv8fGNYn3cowFpLN98L8zO56U0H1PjDIyAlXymgL3Wu7u7v6hbA==
+"@emotion/core@^10.1.1":
+  version "10.1.1"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:1639]: this looks duplicated too

File: yarn.lock [hunk 7]
```diff
@@ -1643,7 +1699,7 @@
     "@emotion/serialize" "^0.11.15"
     "@emotion/utils" "0.11.3"
 
-"@emotion/styled@^10.0.17", "@emotion/styled@^10.0.23":
+"@emotion/styled@^10.0.23":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:1702]: this looks duplicated too

File: yarn.lock [hunk 8]
```diff
@@ -2631,36 +2686,37 @@
...
 "@ledgerhq/errors@^5.22.0":
   version "5.22.0"
   resolved "https://registry.yarnpkg.com/@ledgerhq/errors/-/errors-5.22.0.tgz#7327fc152d4896ddc26aada0943065db21c14880"
   integrity sha512-XDT0meBn39+q+JWzUFXmiFbVYLTy+uHRFMb9napcxyZ0Q/MdKkle9/vkgtvRHjPIkGobklXpyefsgH3BZQHukA==
 
-"@ledgerhq/errors@^5.38.0":
-  version "5.38.0"
-  resolved "https://registry.yarnpkg.com/@ledgerhq/errors/-/errors-5.38.0.tgz#1642b87de47cbabc7b75ca93005a690895920a72"
-  integrity sha512-d4gQzbOLNBoGIwDtEGFNSb0w0aYN10T5Y749e+vqiJoS3dWrB+5BCSQ/U/ALet0wi/UMIyFY/xmgd1gPaPB3Hw==
+"@ledgerhq/errors@^5.48.0":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2704]: this looks duplicated too

File: yarn.lock [hunk 9]
```diff
@@ -2674,15 +2730,15 @@
     "@ledgerhq/logs" "^5.22.0"
     node-hid "^1.3.0"
 
-"@ledgerhq/hw-transport-node-hid-noevents@^5.38.0":
-  version "5.38.0"
-  resolved "https://registry.yarnpkg.com/@ledgerhq/hw-transport-node-hid-noevents/-/hw-transport-node-hid-noevents-5.38.0.tgz#ffbfb0b997e585fc08b6b220997caaa59e094554"
-  integrity sha512-zuxN3gWfCuN+pbK3BKc8z3VCulKI7zee1N3xhCWua9TiwL3leBTBs25Bvm22fmiLJYhzszccbF673/bpnnIQAA==
+"@ledgerhq/hw-transport-node-hid-noevents@^5.48.0":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2733]: this looks duplicated too

File: yarn.lock [hunk 10]
```diff
@@ -2699,19 +2755,19 @@
...
-  resolved "https://registry.yarnpkg.com/@ledgerhq/hw-transport-node-hid/-/hw-transport-node-hid-5.38.0.tgz#db582965fb16b1602430ae664d4e455657eef643"
-  integrity sha512-Daya0ra2NVflHj9ChxhSvf3dJEdxGEWA/qgGA/U/61jJZ5JpvQbzQR6ANSFuXg0IDwWgDea0hsMCviYxd0641A==
-  dependencies:
-    "@ledgerhq/devices" "^5.38.0"
-    "@ledgerhq/errors" "^5.38.0"
-    "@ledgerhq/hw-transport" "^5.38.0"
-    "@ledgerhq/hw-transport-node-hid-noevents" "^5.38.0"
-    "@ledgerhq/logs" "^5.38.0"
-    lodash "^4.17.20"
+"@ledgerhq/hw-transport-node-hid@^5.48.0":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2758]: this looks duplicated too

File: yarn.lock [hunk 11]
```diff
@@ -2743,24 +2799,24 @@
     "@ledgerhq/errors" "^5.22.0"
     events "^3.2.0"
 
-"@ledgerhq/hw-transport@^5.38.0":
-  version "5.38.0"
-  resolved "https://registry.yarnpkg.com/@ledgerhq/hw-transport/-/hw-transport-5.38.0.tgz#b02bea73d77e729d13c367a967df1666f9ef940d"
-  integrity sha512-CAxvHukCcp+RqaEsSltmBw4Lb1yW42fiF/LTYN7JvCkZyLIQhvkndLDVCgp4hpMdtLK4bkf7RJRElqbN0vRoAQ==
+"@ledgerhq/hw-transport@^5.48.0":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2802]: this looks duplicated too

File: yarn.lock [hunk 12]
```diff
@@ -2743,24 +2799,24 @@
...
 "@ledgerhq/logs@^5.22.0":
   version "5.22.0"
   resolved "https://registry.yarnpkg.com/@ledgerhq/logs/-/logs-5.22.0.tgz#a54d6b5b391cdb4c2eacc9500feb04b90475c361"
   integrity sha512-jV4mJxD1aieORm+sK9bYakQd9GMLd7KAxgt2IaxhrTU+QD5Ne47mxQOTys9p7f5w25ujs3R+Px2t3KiMRASHtg==
 
-"@ledgerhq/logs@^5.38.0":
-  version "5.38.0"
-  resolved "https://registry.yarnpkg.com/@ledgerhq/logs/-/logs-5.38.0.tgz#3c5dbd1f62c0bf5580477b218fb57c67b575643a"
-  integrity sha512-i87Yn89Cq2D9Y0KmrEzCm62XHzI2edeOTBENKH6vAyzESGzyF+SBoqtZNwrjJcKup3/9dNn/zHjpicY7ev94Vw==
+"@ledgerhq/logs@^5.48.0":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2816]: this looks duplicated too

File: yarn.lock [hunk 13]
```diff
@@ -2770,27 +2826,27 @@
...
     iterare "1.2.1"
-    tslib "2.0.3"
+    tslib "2.1.0"
     uuid "8.3.2"
 
-"@nestjs/core@7.6.5":
-  version "7.6.5"
-  resolved "https://registry.yarnpkg.com/@nestjs/core/-/core-7.6.5.tgz#93c642d1abf8d3f09f8f78c262814eaf83bc2ad6"
-  integrity sha512-syRpXT09RDMySs1BLQSXJfq1NXGfG4VmF9hZYGef+/QWqTRfSMEDEH5MsCCLt2lK3AZnOXE9BQwWKeNBhKLplA==
+"@nestjs/core@7.6.15":
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:2839]: this looks duplicated too

File: yarn.lock [hunk 14]
```diff
@@ -4696,17 +4670,22 @@
...
-  resolved "https://registry.yarnpkg.com/@webpack-cli/info/-/info-1.2.1.tgz#af98311f983d0b9fce7284cfcf1acaf1e9f4879c"
-  integrity sha512-fLnDML5HZ5AEKzHul8xLAksoKN2cibu6MgonkUj8R9V7bbeVRkd1XbGEGWrAUNYHbX1jcqCsDEpBviE5StPMzQ==
+"@webpack-cli/configtest@^1.0.2":
+  version "1.0.2"
+  resolved "https://registry.yarnpkg.com/@webpack-cli/configtest/-/configtest-1.0.2.tgz#2a20812bfb3a2ebb0b27ee26a52eeb3e3f000836"
+  integrity sha512-3OBzV2fBGZ5TBfdW50cha1lHDVf9vlvRXnjpVbJBa20pSZQaSkMJZiwA8V2vD9ogyeXn8nU5s5A6mHyf5jhMzA==
+
+"@webpack-cli/info@^1.2.3":
+  version "1.2.3"
+  resolved "https://registry.yarnpkg.com/@webpack-cli/info/-/info-1.2.3.tgz#ef819d10ace2976b6d134c7c823a3e79ee31a92c"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:4680]: this looks duplicated too

File: yarn.lock [hunk 15]
```diff
@@ -10200,6 +10046,16 @@ fs-extra@^8.1.0:
     jsonfile "^4.0.0"
     universalify "^0.1.0"
 
+fs-extra@^9.0.0, fs-extra@^9.0.1:
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:10049]: this looks duplicated too

File: yarn.lock [hunk 16]
```diff
@@ -13697,28 +13480,20 @@ node-dir@^0.1.10:
...
-  integrity sha512-NhZ4CsKx7cYm2vSrBAr2PvFOe6sWDf0UYLRqA6svUYg7+/TSfVAu49jYC4BvQ4Sms9SZgdqGBgroqfDhJdTyKQ==
-  dependencies:
-    encoding "^0.1.11"
-    is-stream "^1.0.1"
+node-fetch@2.6.1, node-fetch@^2.6.1:
+  version "2.6.1"
+  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-2.6.1.tgz#045bd323631f76ed2e2b55573394416b639a0052"
+  integrity sha512-V4aYg89jEoVRxRb2fJdAg8FHvI7cEyYdVAh94HH0UIK8oJxUfkjlDQN9RbMx+bEjP7+ggMiFRprSti032Oipxw==
 
 node-fetch@^2.6.0:
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:13488]: this looks duplicated too

File: yarn.lock [hunk 17]
```diff
@@ -13697,28 +13480,20 @@ node-dir@^0.1.10:
...
-node-forge@0.9.0:
-  version "0.9.0"
-  resolved "https://registry.yarnpkg.com/node-forge/-/node-forge-0.9.0.tgz#d624050edbb44874adca12bb9a52ec63cb782579"
-  integrity sha512-7ASaDa3pD+lJ3WvXFsxekJQelBKRpne+GOVbLbtHYdd7pFspyeuJHnWfLplGf3SwKGbfs/aYl5V/JCIaHVUKKQ==
+node-forge@^0.10.0:
+  version "0.10.0"
+  resolved "https://registry.yarnpkg.com/node-forge/-/node-forge-0.10.0.tgz#32dea2afb3e9926f02ee5ce8794902691a676bf3"
+  integrity sha512-PPmu8eEeG9saEUvI97fm4OYxXVB6bFvyNTyiUOBichBpFG8A1Ljw3bY62+5oOjDEMHRnd0Y7HQ+x7uzxOzC6JA==
 
 node-forge@^0.7.6:
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:13498]: this looks duplicated too

File: yarn.lock [hunk 18]
```diff
@@ -15890,6 +15646,16 @@ rc-collapse@~3.1.0:
...
+  version "8.5.2"
+  resolved "https://registry.yarnpkg.com/rc-dialog/-/rc-dialog-8.5.2.tgz#530e289c25a31c15c85a0e8a4ba3f33414bff418"
+  integrity sha512-3n4taFcjqhTE9uNuzjB+nPDeqgRBTEGBfe46mb1e7r88DgDo0lL4NnxY/PZ6PJKd2tsCt+RrgF/+YeTvJ/Thsw==
+  dependencies:
+    "@babel/runtime" "^7.10.1"
+    classnames "^2.2.6"
+    rc-motion "^2.3.0"
+    rc-util "^5.6.1"
+
 rc-dialog@~8.5.1:
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:15659]: this looks duplicated too

File: yarn.lock [hunk 19]
```diff
@@ -16083,6 +15848,19 @@ rc-select@^12.0.0, rc-select@~12.1.0:
     rc-util "^5.0.1"
     rc-virtual-list "^3.2.0"
 
+rc-select@~12.1.6:
+  version "12.1.7"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:15852]: this looks duplicated too

File: yarn.lock [hunk 20]
```diff
@@ -16146,14 +15924,22 @@ rc-textarea@^0.3.0, rc-textarea@~0.3.0:
...
+rc-tooltip@^5.0.1:
   version "5.0.1"
   resolved "https://registry.yarnpkg.com/rc-tooltip/-/rc-tooltip-5.0.1.tgz#b82c4259604d2cb62ca610ed7932dd37fc6ef61d"
   integrity sha512-3AnxhUS0j74xAV3khrKw8o6rg+Ima3nw09DJBezMPnX3ImQUAnayWsPSlN1mEnihjA43rcFkGM1emiKE+CXyMQ==
   dependencies:
     "@babel/runtime" "^7.11.2"
     rc-trigger "^5.0.0"
 
+rc-tooltip@~5.1.0:
+  version "5.1.0"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:15936]: this looks duplicated too

File: yarn.lock [hunk 21]
```diff
@@ -16213,6 +16010,15 @@ rc-util@^5.5.1, rc-util@^5.6.1, rc-util@^5.6.2, rc-util@^5.6.6:
     react-is "^16.12.0"
     shallowequal "^1.1.0"
 
+rc-util@^5.7.0, rc-util@^5.8.0, rc-util@^5.9.4:
+  version "5.9.8"
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:16014]: this looks duplicated too

File: yarn.lock [hunk 22]
```diff
@@ -17334,7 +17110,14 @@ run-queue@^1.0.0, run-queue@^1.0.3:
...
 
-rxjs@6.6.3, rxjs@^6.5.2, rxjs@^6.5.3, rxjs@^6.6.0, rxjs@^6.6.2, rxjs@^6.6.3:
+rxjs@6.6.7, rxjs@^6.6.7:
+  version "6.6.7"
+  resolved "https://registry.yarnpkg.com/rxjs/-/rxjs-6.6.7.tgz#90ac018acabf491bf65044235d5863c4dab804c9"
+  integrity sha512-hTdwr+7yYNIT5n4AMYp85KA6yw2Va0FLa3Rguvbpa4W3I5xynaBZo41cM3XM+4Q6fRMj3sBYIR1VAmZMXYJvRQ==
+  dependencies:
+    tslib "^1.9.0"
+
+rxjs@^6.5.2, rxjs@^6.5.3, rxjs@^6.6.0, rxjs@^6.6.2, rxjs@^6.6.3:
```
Comment (thatStrangeGuyThorchain) [inline yarn.lock:17120]: this looks duplicated too

**Comment by veado** [comment]
> pls double-check duplicated deps

Not sure if all of these really duplications. There are no issues with it, but a risk to run into another dependency issues. That's why I will ignore it for now.

**Pull request merged**